### PR TITLE
Fix calculation of 'end of previous work week'

### PR DESCRIPTION
**Pull request by lauft**
The calculation of the _end of previous work week_ (eopww) fails if the current day is a Saturday.
The algorithm ([Datetime.cpp:2059](https://github.com/GothenburgBitFactory/libshared/blob/ee050a2cc68c46e8748113246b26feb0ad9f34c5/src/Datetime.cpp#L2059-L2063)) subtracts the current weekday (with Sunday = 0) plus one modulo 7 from the current day of month.

When the current day is a Saturday, this results in a subtraction of 0, i.e. the current day remains unchanged, while it should be the previous Saturday.

Dropping the modulo fixes the issue.

### CODE REVIEW

**Review by djmitche — APPROVED** [review]

File: test/dates.t.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ void testInit (UnitTest& t, const std::string& value, Datetime& var)
 ////////////////////////////////////////////////////////////////////////////////
 int main (int, char**)
 {
-  UnitTest t (164);
```
Comment (djmitche) [inline test/dates.t.cpp:55]: Hang on, this just changes the number of expected tests. Let's add a test case for this?

**Review by djmitche — CHANGES_REQUESTED** [review]

File: test/dates.t.cpp (hunk 1 above)
Comment (lauft) [reply test/dates.t.cpp:55]: The currently failing testcase is [here](https://github.com/GothenburgBitFactory/libshared/blob/ee050a2cc68c46e8748113246b26feb0ad9f34c5/test/dates.t.cpp#L239):
```
t.ok (eopww < soww, "eopww < soww");
```
in `dates.t.cpp`. It only fails on a Saturday though...

Comment (lauft) [reply test/dates.t.cpp:55]: The change her only aligns the number of actually present testcases with the number of planned ones (i.e. the argument of the `UnitTest` constructor).

Comment (lauft) [reply test/dates.t.cpp:55]: Having a specific test case for "`eopww` should be before `soww` on Saturdays" would require the reference date (a.k.a. `now`) to be injectable into the Datetime parser, which would be preferable, but a bigger endeveaour... 🤔

Comment (djmitche) [reply test/dates.t.cpp:55]: Ah, I see -- weird to have date-time tests depend on the _actual_ date and time. Yeah, fuzzing that (or just replacing them with fake time and figuring out the right times to test all the corner cases) sounds like a good, but larger project.

**Review by djmitche — APPROVED** [review]
